pipeline.ml

```diff
@@ -69,27 +63,39 @@ let pipeline ~github ~repo ~output_file ?slack_path () =
             "exec";
             "--";
             "bench/db_bench.exe";
-            "-b";
+            "--bench";
             "index";
-            "-j";
-            output_file;
+            "--json";
+            Fmt.str "%a" Fpath.pp tmp_target;
           ]
     in
+    (* Conditionally move the results to ?output_file *)
+    let results_path =
+      match output_file with
+      | Some path ->
+          Bos.OS.Path.move tmp_source path |> Rresult.R.error_msg_to_invalid_arg;
+          path
+      | None -> tmp_source
+    in
     (* No need to read JSON if we're not publishing the results anywhere *)
-    match slack_path with Some _ -> Some (read_json tmp_source) | None -> None
+    match slack_path with
+    | Some p -> Some (p, read_fpath results_path)
+    | None -> None
   in
   s
-  |> Current.option_map (fun results ->
-         let channel = read_channel_uri (Option.get slack_path) in
-         Slack.post channel ~key:"output" results)
+  |> Current.option_map (fun p ->
+         Current.component "post"
+         |> let** path, _ = p in
```

Comment:
This bit is more complicated than it probably should be... I'm trying to do something about that on the OCurrent side of things :slightly_smiling_face: This should do for now.